### PR TITLE
Don't include requirements only needed for example pipeline

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Description

context: #3386 

## Development notes

- Updated `utils.py` to remove `seaborn` and `scikit-learn` and the extras from `kedro-datasets`
- Updated requirements and toml to `Path` parameter types
- Updated release notes

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [x] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/templates/project/hooks/utils.py [hunk 1]
```diff
@@ -172,6 +198,10 @@ def setup_template_tools(selected_tools_list: str, requirements_file_path: str,
 
     if ("PySpark" in selected_tools_list or "Kedro Viz" in selected_tools_list) and example_pipeline != "True":
         _remove_pyspark_viz_starter_files("Kedro Viz" in selected_tools_list, python_package_name)
+        # Remove requirements used by example pipelines
+        _remove_from_file(requirements_file_path, example_pipeline_requirements)
+        _remove_extras_from_kedro_datasets(requirements_file_path)
```
Comment (merelcht) [inline kedro/templates/project/hooks/utils.py:203]: We also need to do this when there's no example chosen, but also no `pyspark` or viz right? https://github.com/kedro-org/kedro-starters/blob/main/spaceflights-pandas/%7B%7B%20cookiecutter.repo_name%20%7D%7D/requirements.txt

File: kedro/templates/project/hooks/utils.py [hunk 2]
```diff
@@ -141,13 +144,36 @@ def _remove_pyspark_viz_starter_files(is_viz: bool, python_package_name: str) ->
...
+def _remove_extras_from_kedro_datasets(file_path: Path) -> None:
+    """Remove all extras from kedro-datasets in the requirements file.
+
+    Args:
+        file_path (Path): The path of the requirements file.
+    """
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    new_lines = []
```
Comment (DimedS) [inline kedro/templates/project/hooks/utils.py:156]: Thanks for PR, @SajidAlamQB ! I have a small question, is lines a List? If yes why you are creating a new List new_lines and copying, not just modifying 1 row in list variable?

Comment (SajidAlamQB) [reply kedro/templates/project/hooks/utils.py:156]: It was for readability but can amend for simplification.

File: kedro/templates/project/hooks/utils.py [hunk 3]
```diff
@@ -15,6 +15,9 @@
 # Configuration key for documentation dependencies
 docs_pyproject_requirements = ["project.optional-dependencies"]  # For pyproject.toml
 
+# Requirements for example pipelines
+example_pipeline_requirements = "seaborn~=0.12.1\nscikit-learn~=1.0\n"
```
Comment (DimedS) [inline kedro/templates/project/hooks/utils.py:19]: This approach seems somewhat risky, as we are hardcoding the complete version names for all the libraries slated for deletion. I'm concerned about how we will remember to update these if we decide to modify the version requirements in the future. This is also for lines 8-12.

**Review by DimedS — APPROVED** [review]
Thank you for PR, @SajidAlamQB . Great job! Left few minor comments.

File: kedro/templates/project/hooks/utils.py (hunk 3 above)
Comment (SajidAlamQB) [reply kedro/templates/project/hooks/utils.py:19]: Valid point I had the same feelings but we'll need to look for a more robust way to do it, maybe with regex?

File: kedro/templates/project/hooks/utils.py (hunk 2 above)
Comment (SajidAlamQB) [reply kedro/templates/project/hooks/utils.py]: Simplified.

File: kedro/templates/project/hooks/utils.py (hunk 3 above)
Comment (SajidAlamQB) [reply kedro/templates/project/hooks/utils.py:19]: I can create a follow up issue on this.

File: kedro/templates/project/hooks/utils.py [hunk 4]
```diff
@@ -141,13 +144,36 @@ def _remove_pyspark_viz_starter_files(is_viz: bool, python_package_name: str) ->
...
+        file_path (Path): The path of the requirements file.
+    """
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    new_lines = []
+    for line in lines:
+        if 'kedro-datasets[' in line:
+            # Remove the extras by keeping only the part before the '['
+            new_line = line.split('[')[0] + '\n'
```
Comment (DimedS) [inline kedro/templates/project/hooks/utils.py]: you are removing version on the end, is it ok?

File: kedro/templates/project/hooks/utils.py (hunk 3 above)
Comment (DimedS) [reply kedro/templates/project/hooks/utils.py:19]: I agree, we need to think more about it, issue is a good point 👍

File: kedro/templates/project/hooks/utils.py (hunk 4 above)
Comment (SajidAlamQB) [reply kedro/templates/project/hooks/utils.py]: Good spot made a change to keep it.

File: kedro/templates/project/hooks/utils.py (hunk 3 above)
Comment (SajidAlamQB) [reply kedro/templates/project/hooks/utils.py:19]: Done.
https://github.com/kedro-org/kedro/issues/3432

File: kedro/templates/project/hooks/utils.py (hunk 1 above)
Comment (merelcht) [reply kedro/templates/project/hooks/utils.py:202]: But `scikit-learn` is also a requirement in `spaceflights-pandas` as I've linked above.

Comment (merelcht) [reply kedro/templates/project/hooks/utils.py:202]: Ah yes you're right!

**Review by merelcht — APPROVED** [review]
LGTM! 👍

**Pull request merged**